sys/include/net/ng_ipv6/netif.h

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    net_ng_ipv6_netif IPv6 network interfaces
+ * @ingroup     net_ng_ipv6
+ * @brief       IPv6 specific information on @ref net_ng_netif
+ * @{
+ *
+ * @file
+ * @brief       Definitions for IPv6 specific information of network interfaces.
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#ifndef NG_IPV6_NETIF_H_
+#define NG_IPV6_NETIF_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "kernel_macros.h"
+#include "kernel_types.h"
+#include "mutex.h"
+#include "net/ng_ipv6/addr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @def NG_IPV6_NETIF_ADDR_NUMOF
+ *
+ * @brief   Number of IPv6 addresses per interface.
+ */
+#ifdef MODULE_NG_IPV6_ROUTER
+#define NG_IPV6_NETIF_ADDR_NUMOF    (5) /* router needs all-routers multicast address */
+#else
+#define NG_IPV6_NETIF_ADDR_NUMOF    (4)
```

Comment:
Is this a limit on the maximum number of addresses or is it the default number of addresses when booting?

Don't forget that address autoconfiguration may add more addresses (IPv6-RA, RPL). It must be possible to configure at least one global address after all of the multicast and link-local addresses have been configured.